contracts/ReputationMiningCycle.sol

```diff
@@ -188,7 +190,8 @@ contract ReputationMiningCycle is ReputationMiningCycleStorage, PatriciaTreeProo
     IColonyNetwork(colonyNetworkAddress).setReputationRootHash(
       submission.proposedNewRootHash,
       submission.nNodes,
-      submittedHashes[submission.proposedNewRootHash][submission.nNodes]
+      submittedHashes[submission.proposedNewRootHash][submission.nNodes],
+      1200 * WAD // TODO: Make this a function of reputation state
```

Comment:
Can we avoid going round the houses and passing the `reward` to `IColonyNetwork. setReputationRootHash ` which calls `IColonyNetwork.rewardStakers` which in turns circles back to here `IReputationMiningCycle.rewardStakersWithReputation` and make it a separate internal contract call?

Reply:
If that 👆 works we'll avoid importing `ITokenLocking` in `ColonyNetworkMining` too and reuse the import here.

Reply:
Rewarding miners with tokens can stay in `ColonyNetworkMining` but reputation rewards can likely be done here.

Reply:
Hmm. I would say in that case let's move the whole reward flow into `ReputationMiningCycle`. Having the reward logic split over two contracts seems unnecessary; we can get `metaColonyAddress` via `IColonyNetwork(colonyNetworkAddress)`.

Reply:
Erm wait, I'm noticing that we're not technically going in a circle -- the original contract is the `activeReputationMiningCycle` and we call back into the `inactiveReputationMiningCycle`. So this circuitous route is not so circuitous after all.

Reply:
You're right. I still experimented a little with refactoring this but got no significant benefits clarity wise. Considering `confirmNewHash` gas cost of 687,823 which is less than a dollar at current prices this isn't a priority atm so happy to leave as is, for now at least.